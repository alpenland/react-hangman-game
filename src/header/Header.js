import React from 'react';
import styled from 'styled-components';

const StyledHeader = styled.h1`
    color: #1895ac;
`

const Header = () => {
    return (
        <div>
            <StyledHeader>Hangman</StyledHeader>
            <p>
                Choose a category. Guess one letter at a time to figure out the word or phrase. <br />
                You have 7 lives. Each wrong guess costs 1 life. Good luck! <br /><br />
                Choose a new category to start over or play again.    
            </p>
        </div>
    )
};

export default Header;