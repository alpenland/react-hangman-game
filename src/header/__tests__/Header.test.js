import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Header />);
});

it('should render Header component', () => {
    expect(wrapper).toMatchSnapshot();
});

it('should contian 1 h2, 1 p tag', () => {
    expect(wrapper.find('h2').length).toBe(1);
    expect(wrapper.find('p').length).toBe(1);
});

