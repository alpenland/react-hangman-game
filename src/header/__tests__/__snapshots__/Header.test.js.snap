// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Hangman
        </h2>,
        <p>
          Choose a category. Guess one letter at a time to figure out the word or phrase. You have 7 lives. Each wrong guess costs 1 life. Good luck!
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hangman",
        },
        "ref": null,
        "rendered": "Hangman",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Choose a category. Guess one letter at a time to figure out the word or phrase. You have 7 lives. Each wrong guess costs 1 life. Good luck!",
        },
        "ref": null,
        "rendered": "Choose a category. Guess one letter at a time to figure out the word or phrase. You have 7 lives. Each wrong guess costs 1 life. Good luck!",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Hangman
          </h2>,
          <p>
            Choose a category. Guess one letter at a time to figure out the word or phrase. You have 7 lives. Each wrong guess costs 1 life. Good luck!
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hangman",
          },
          "ref": null,
          "rendered": "Hangman",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose a category. Guess one letter at a time to figure out the word or phrase. You have 7 lives. Each wrong guess costs 1 life. Good luck!",
          },
          "ref": null,
          "rendered": "Choose a category. Guess one letter at a time to figure out the word or phrase. You have 7 lives. Each wrong guess costs 1 life. Good luck!",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
